src/components/HomePageContent/HomePageContainer.tsx

```diff
@@ -19,21 +18,28 @@ const { error } = getLogger("component:mainpage");
 
 export const HomePageContainer = (): React.ReactElement => {
   const router = useRouter();
-  const history = useHistory();
   const dispatch = useDispatch();
   const loadCertificate = React.useCallback((payload: any) => dispatch(updateCertificate(payload)), [dispatch]);
   // event listener for any custom postMessage
-  window.addEventListener("message", (event) => {
-    if (event.data.type === NestedDocumentState.LOAD) {
-      try {
-        const doc = atob(event.data.payload);
-        loadCertificate(JSON.parse(doc));
-        history.push("/verify");
-      } catch (e) {
-        error("decode data not json: " + e);
+
+  useEffect(() => {
```

Comment:
it's weird to see useEffect here, and React.useEffect below ^^ (I have no opinion on what you should use)

Reply:
will clean this